src/parallel-rustc.md

```diff
@@ -48,9 +124,12 @@ When a query `foo` is evaluated, the cache table for `foo` is locked.
   start evaluating.
 - If there *is* another query invocation for the same key in progress, we
   release the lock, and just block the thread until the other invocation has
-  computed the result we are waiting for. This cannot deadlock because, as
-  mentioned before, query invocations form a DAG. Some threads will always make
-  progress.
+  computed the result we are waiting for. **Deadlocks are possible**, in which
+  case `rustc_query_system::query::job::deadlock()` will be called to detect
```

Comment:
I think I wouldn't call this a deadlock per-se, just cycle detection in a parallel setting. It's possible that the compiler has bugs and actually deadlocks but in general we should never *actually* deadlock: i.e., stop doing anything with multiple threads waiting on each other.